### PR TITLE
[service.subtitles.wizdom] 2.0.2

### LINKED ISSUE #1059: [Update] service.subtitles.wizdom
### Add-on details:
- **General**
  - Add-on name: Wizdom Subs
  - Add-on ID: service.subtitles.wizdom
  - Version number: 1.0.7
  - Kodi/repository version: krypton, leia
  - URL: https://github.com/wizupdate/service.subtitles.wizdom
  - Revision/tag: 1.0.7
  - Branch: master

### Description of changes:
Source URL change from the initial release of this add-on.

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### PR DESCRIPTION
**Pull request by enen92**
Requested by the author in https://github.com/xbmc/repo-scripts/issues/1059

closes #1059

### CODE REVIEW

File: service.subtitles.wizdom/service.py [hunk 1]
```diff
@@ -1,297 +1,345 @@
...
-__subs__ = unicode(translatePath(path.join(__temp__, 'subs')), 'utf-8')
-__scriptname__ = __addon__.getAddonInfo('name')
-__language__ = __addon__.getLocalizedString
-
+from xbmc import translatePath, executebuiltin, getInfoLabel, executeJSONRPC, Player, log, getCondVisibility
+
+myAddon = Addon()
+myScriptID = myAddon.getAddonInfo('id')
+myVersion = myAddon.getAddonInfo('version')
+myTmp = translatePath(myAddon.getAddonInfo('profile')).encode('utf-8')
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: encode('utf-8') is not required in the python3 version. Maybe use kodi-six?

File: service.subtitles.wizdom/service.py [hunk 2]
```diff
@@ -1,297 +1,345 @@
...
+mySubFolder = translatePath(path.join(myTmp, 'subs')).encode('utf-8')
+myName = myAddon.getAddonInfo('name')
+myLang = myAddon.getLocalizedString
+
+def getDomain():
+	try:
+		myDomain = str(get('https://pastebin.com/raw/1vbRPSGh').text)
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: sorry this is not allowed. Either hardcode the url or move it to a setting. No pastebin shaddy links please.
If you control the domain manage the destination in your DNS configuration not in kodi.

File: service.subtitles.wizdom/service.py [hunk 3]
```diff
@@ -1,297 +1,345 @@
...
+	filename = 'wizdom.imdb.%s.%s.%s.json' % (imdb, season, episode)
+	url = "http://json.%s/search.php?action=by_id&imdb=%s&season=%s&episode=%s&version=%s" % (
+		myDomain, imdb, season, episode, version)
+
+	wlog("searchByIMDB:%s" % url)
+	json = cachingJSON(filename,url)
+	subs_rate = []  # TODO remove not in used
+	if json != 0:
+		for item_data in json:
+			listitem = ListItem(label="Hebrew", label2=item_data["versioname"],
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: drop thumbnailImage and iconImage kwargs. Use only setArt:

`listitem.setArt({'thumb': xxx, 'icon': xxx})`

File: service.subtitles.wizdom/service.py [hunk 4]
```diff
@@ -1,297 +1,345 @@
...
+		if item['episode'] == '' or item['episode'] < 1:
+			item['episode'] = 0
+
+		if item['episode'] == 0:
+			item['title'] = lowercase_with_underscores(getInfoLabel("VideoPlayer.Title"))  # no original title, get just Title
+		else:
+			item['title'] = lowercase_with_underscores(getInfoLabel("VideoPlayer.TVshowtitle"))  # Show
+		if item['title'] == "":
+			item['title'] = lowercase_with_underscores(getInfoLabel("VideoPlayer.OriginalTitle"))  # try to get original title
+		item['file_original_path'] = unquote(unicode(Player().getPlayingFile(), 'utf-8'))  # Full path of a playing file
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: won't work in python3

File: service.subtitles.wizdom/service.py [hunk 5]
```diff
@@ -1,297 +1,345 @@
...
+
+	# Search Local File
+	if not imdb_id:
+		ManualSearch(item['title'])
+	endOfDirectory(int(sys.argv[1]))
+	if myAddon.getSetting("Debug") == "true":
+		if imdb_id[:2] == "tt":
+			Dialog().ok("Debug "+myVersion, str(item), "imdb: "+str(imdb_id))
+		else:
+			Dialog().ok("Debug "+myVersion, str(item), "NO IDS")
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: NO IDS string needs to be localized as it is presented to the user. Same goes for the "Debug" string.

File: service.subtitles.wizdom/service.py [hunk 6]
```diff
@@ -1,297 +1,345 @@
...
+	# Upload AP
+	try:
+		if urlparse(Player().getPlayingFile()).hostname[-11:]=="tv4.live":
+			Ap = 1
+	except:
+		pass
+	
+	if Ap==1 and myAddon.getSetting("uploadAP") == "true":
+		try:
+			response = get("http://subs.vpnmate.com/webupload.php?status=1&imdb=%s&season=%s&episode=%s"%(getParam("imdb", params),getParam("season", params),getParam("episode", params)))
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: what the hell is this? What is vpnmate?

File: service.subtitles.wizdom/service.py [hunk 7]
```diff
@@ -1,297 +1,345 @@
...
+			ap_object = loads(response.text)["result"]
+			if ap_object["lang"]["he"]==0:
+				xbmc.sleep(30*1000)
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: sorry, if you want to sleep for 30 seconds you must use xbmc.Monitor:

`xbmc.Monitor().waitForAbort(30)`

Your current approach will hang kodi when exiting.

File: service.subtitles.wizdom/service.py [hunk 8]
```diff
@@ -1,297 +1,345 @@
...
+				i = Dialog().yesno("Apollo Upload Subtitle","Media version %s"%ap_object["version"],"This subtitle is 100% sync and match?")
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: needs to be localized

File: service.subtitles.wizdom/service.py [hunk 9]
```diff
@@ -1,297 +1,345 @@
...
-        rmtree(__temp__)
-    except:
-        pass
-    executebuiltin((u'Notification(%s,%s)' % (__scriptname__, __language__(32004))).encode('utf-8'))
+	try:
+		rmtree(myTmp)
+	except Exception as err:
+		wlog('Caught Exception: deleting tmp dir: %s' % format(err))
+		pass
+	executebuiltin((u'Notification(%s,%s)' % (myName, myLang(32004))).encode('utf-8'))
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: xbmcgui.Dialog().notification() use the python API and not builtins

File: service.subtitles.wizdom/service.py [hunk 10]
```diff
@@ -1,297 +1,345 @@
...
+	
+	if Ap==1 and myAddon.getSetting("uploadAP") == "true":
+		try:
+			response = get("http://subs.vpnmate.com/webupload.php?status=1&imdb=%s&season=%s&episode=%s"%(getParam("imdb", params),getParam("season", params),getParam("episode", params)))
+			ap_object = loads(response.text)["result"]
+			if ap_object["lang"]["he"]==0:
+				xbmc.sleep(30*1000)
+				i = Dialog().yesno("Apollo Upload Subtitle","Media version %s"%ap_object["version"],"This subtitle is 100% sync and match?")
+				if i == 1:
+					response = get("http://subs.vpnmate.com/webupload.php?upload=1&lang=he&subid=%s&imdb=%s&season=%s&episode=%s"%(getParam("id", params),getParam("imdb", params),getParam("season", params),getParam("episode", params)))
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: same thing. What the hell is this address?

File: service.subtitles.wizdom/service.py [hunk 11]
```diff
@@ -1,297 +1,345 @@
...
+					ap_upload = loads(response.text)["result"]
+					if "error" in ap_upload:
+						Dialog().ok("Apollo Error","%s"%ap_upload["error"])
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: localize please

File: service.subtitles.wizdom/service.py [hunk 12]
```diff
@@ -1,297 +1,345 @@
...
+					else:
+						Dialog().ok("Apollo","Sub uploaded. Thank you!")
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: same

File: service.subtitles.wizdom/service.py [hunk 13]
```diff
@@ -1,297 +1,345 @@
...
+	subs = download(id)
+	for sub in subs:
+		listitem = ListItem(label=sub)
+		addDirectoryItem(handle=int(sys.argv[1]), url=sub, listitem=listitem, isFolder=False)
+	endOfDirectory(int(sys.argv[1]))
+	Ap = 0
+
+	# Upload AP
+	try:
+		if urlparse(Player().getPlayingFile()).hostname[-11:]=="tv4.live":
```
Comment (enen92) [inline service.subtitles.wizdom/service.py]: what's the use case for this? What is tv4.live?

**Comment by enen92** [comment]
See also travis failures:

```
ERROR: Image icon should be explicitly declared in addon.xml <assets>.
ERROR: Icon.png should be solid. It has transparency.
ERROR: Icon should have either 256x256 or 512x512 but it has 218x256
```

```
ERROR: Using the old language directory structure, please move to the new one.
ERROR: Invalid PO file ./service.subtitles.wizdom/resources/language/English/strings.po: Syntax error on line 15
ERROR: Invalid PO file ./service.subtitles.wizdom/resources/language/Hebrew/strings.po: Syntax error on line 15
```

**Comment by wizupdate** [comment]
Thanks, We will work on a fix.

**Comment by enen92** [comment]
More than 1 month passed. I don't like to have my PRs open for too long. Closing.

**Pull request closed without merging**